Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.4-slim
+WORKDIR /app
+COPY ./images /app/images
+COPY ./libgrabsite /app/libgrabsite
+COPY ./grab-site ./gs-dump-urls ./gs-server ./setup.py /app/
+RUN apt-get update
+RUN apt-get install -y build-essential
+RUN pip3 install ./
+RUN apt-get purge -y build-essential
+RUN apt-get autoremove -y
+RUN apt-get clean
```

Comment:
Good point! I combined the commands & got the image size down to 235.4MB. I'm kinda surprised that images don't get flattened, but https://github.com/docker/docker/issues/332 offers a lengthy discussion on it.

As for basing it on `python:3.4`, that would reduce the build time & total size of images on the system, but only if a significant number of the other images on the system are based on it too, which I don't think we can assume. It's probably better to just optimize for the smallest resulting image size.